lib/Interpreter/DeclUnloader.cpp

```diff
@@ -649,7 +649,12 @@ bool DeclUnloader::VisitRedeclarable(clang::Redeclarable<T>* R, DeclContext* DC)
     GD = GlobalDecl(CXXDtor, Dtor_Base);
     MaybeRemoveDeclFromModule(GD);
     GD = GlobalDecl(CXXDtor, Dtor_Comdat);
+#if defined(LLVM_ON_WIN32)
+    // MicrosoftMangle.cpp:954 calls llvm_unreachable when mangling Dtor_Comdat
```

Comment:
Can you check for this in the `MaybeRemoveDeclFromModule`, checking for such kinds on declarations and skipping them?